plans/destroy.pp

```diff
@@ -16,17 +16,24 @@
 plan provision::destroy(
   Provision::CloudProvider  $provider      = 'aws',
   String[1]                 $resource_name = 'puppetlabs-provision',
-  String[1]                 $profile       = undef,
   Optional[String[1]]       $region        = 'us-west-2',
+  Optional[Hash[String[1], String[1]]] $provider_options = undef,
 ) {
   $_tf_dir = "terraform/${provider}/"
 
+  if $provider == 'aws' {
```

Comment:
Can we solve provider-specific stuff either in the terraform layer or in the terraform plan layer?

Reply:
ahh I think I misplaced it while testing it, it supposed to be in cloud specific checks into sub plan.